### PR TITLE
Reject transactions for which there’s no chance for the chain to reach the required gas price

### PR DESCRIPTION
**Pull request by rafal-ch**
This PR adds another check to the transaction acceptor.

It makes the acceptor reject transactions with gas price tolerance set to a value that is not achievable by the chain within the given TTL.

Calculation is based on the assumptions that chain gas price can only go up and down by 1 every each era.

Additionally, this PR also fixes the problem with the gas price being reset after the network upgrade. With the fix, the gas price is preserved across the upgrades, which is covered by a new NCTL upgrade test in the [corresponding PR](https://github.com/casper-network/casper-nctl/pull/14).

### CODE REVIEW

**Comment by rafal-ch** [comment]
bors try

**Comment by rafal-ch** [comment]
bors try

File: types/src/block/block_header.rs [hunk 1]
```diff
@@ -174,6 +174,14 @@ impl BlockHeader {
         }
     }
 
+    /// Returns the gas price for the upcoming era (if this is a switch block).
+    pub fn next_era_gas_price(&self) -> Option<u8> {
+        match self {
+            BlockHeader::V1(_) => None,
```
Comment (EdHastingsCasperAssociation) [inline types/src/block/block_header.rs:180]: this could just be 1

**Comment by devendran-m** [comment]
Moving it back to backlog

**Comment by EdHastingsCasperAssociation** [comment]
After some back and forth deliberation, we decided to not include this in 2.0 rel

**Pull request closed without merging**